gnomad_qc/v5/annotations/generate_frequency.py

```diff
@@ -0,0 +1,445 @@
+        qual_hists=aou_variant_mt.qual_hists.annotate(
+            **all_sites_an_ht[aou_variant_mt.row_key].qual_hists,
+        )
+    )
+    logger.info("Annotating frequencies with all sites ANs...")
+    group_membership_ht = group_membership(test=test, data_set="aou").ht()
+    aou_variant_freq_ht = agg_by_strata(
+        aou_variant_mt.select_entries(
+            "GT",
+            "adj",
+            n_alt_alleles=aou_variant_mt.GT.n_alt_alleles(),
+            is_hom_var=aou_variant_mt.GT.is_hom_var(),
+        ),
+        {
+            "AC": (lambda t: t.n_alt_alleles, hl.agg.sum),
+            "homozygote_count": (lambda t: t.is_hom_var, hl.agg.count_where),
+        },
+        group_membership_ht=group_membership_ht,
+        select_fields=["hists_fields"],
+    )
+
+    # Load AN values from consent_ans (calculated by another script but used
+    # same group membership HT so same strata order)
+    logger.info("Loading AN values from consent_ans...")
+    aou_variant_freq_ht = aou_variant_freq_ht.annotate(
+        all_sites_an=all_sites_an_ht[aou_variant_freq_ht.key].AN
+    )
+
+    logger.info("Building complete frequency struct with imported AN values...")
+    aou_variant_freq_ht = aou_variant_freq_ht.annotate(
+        freq=hl.map(
+            lambda freq_data, an_data: hl.struct(
+                AC=freq_data.AC,
+                AF=hl.if_else(an_data > 0, freq_data.AC / an_data, 0.0),
+                AN=an_data,
+                homozygote_count=freq_data.homozygote_count,
+            ),
+            aou_variant_freq_ht.freq,
+            aou_variant_freq_ht.all_sites_an,
+        ),
+    ).drop("all_sites_an")
+
+    # Nest histograms to match gnomAD structure
+    # Note: hists_fields.qual_hists already contains raw_qual_hists and qual_hists
+    # as nested fields due to split_adj_and_raw=True in qual_hist_expr
+    aou_variant_freq_ht = aou_variant_freq_ht.select(
+        freq=aou_variant_freq_ht.freq,
+        histograms=hl.struct(
+            qual_hists=aou_variant_freq_ht.hists_fields.qual_hists.qual_hists,
+            raw_qual_hists=aou_variant_freq_ht.hists_fields.qual_hists.raw_qual_hists,
+            age_hists=aou_variant_freq_ht.hists_fields.age_hists,
+        ),
+    )
+
+    return aou_variant_freq_ht
+
+
+def _calculate_aou_frequencies_and_hists_using_densify(
+    aou_vds: hl.vds.VariantDataset,
+    test: bool = False,
+) -> hl.Table:
+    """
+    Calculate frequencies and age histograms for AoU variant data using densify.
+
+    :param aou_vds: Prepared AoU VariantDataset
+    :param test: Whether to use test resources
+    :return: Table with freq and age_hists annotations
+    """
+    logger.info("Annotating quality metrics histograms and age histograms...")
+    aou_mt = hl.vds.to_dense_mt(aou_vds)
+    aou_mt = aou_mt.annotate_rows(hists_fields=mt_hists_fields(aou_mt))
+    aou_freq_ht = compute_freq_by_strata(aou_mt, select_fields=["hists_fields"])
```

Comment:
freq_meta_sample_count  counts look too high

Reply:
I can look into this. The aou_vds function is called with release only so maybe something is off there.

Reply:
I'm seeing 365318 for adj/raw for both methods, is that what you are refering to? Or the gen anc groups?